components/widgets/ui/src/main/resources/PhenoTips/GeneVariantMacros.xml

```diff
@@ -0,0 +1,1420 @@
+      Event.fire(document, 'xwiki:dom:updated', {'elements' : [newGeneRow]});
+    },
+    ajaxDeleteVariantData : function (event) {
+      event.stop();
+      var deleteTrigger = event.element();
+      var geneClass = this.geneClassName;
+      if (deleteTrigger.disabled) {
+         return;
+      }
+      new XWiki.widgets.ConfirmedAjaxRequest(deleteTrigger.href, {
+        onCreate : function() {
+          deleteTrigger.disabled = true
+        },
+        onSuccess : function() {
+          var dataRow = deleteTrigger.up('tr:not(.head-group)');
+          if (dataRow) {
+            var className = dataRow.previous().className;
+            var geneIndex = parseInt(className.substring(className.lastIndexOf('-')+1));
+            //we are deleting the only variant in the gene: remove variant table header, 
+            //  unlock the genesymbol input field
+            if (dataRow.previous().hasClassName('variant-title-row') &amp;&amp; dataRow.next(1).hasClassName('variant-footer')) {
+              $$('.variant-gene-'+geneIndex+'.variant-title-row').each(function(item) {
+                item.remove();
+              });
+              $$('[name="'+geneClass+'_' + geneIndex + '_gene"]')[0].toggleClassName('v-collapsed', false);
+              $$('.gene.col-label.gene-' + geneIndex + '.gene-input-label')[0].toggleClassName('v-collapsed', true);
+              dataRow.next().remove();
+              dataRow.remove();
+            } else {
+              //we are removing variant from the middle of the table-> updating table count
+              dataRow.next().remove();
+              dataRow.remove();
+              var innerText = deleteTrigger.up('tr').select('.variant-row-count.variant')[0].innerText;
+              var geneCount = innerText.substring(0, innerText.indexOf('.'));
+              var i = 1;
+              $$('.variant-hide-heading-' + geneIndex + ' .variant-row-count').each(function(item) {
+                item.childNodes[1].textContent = geneCount + '.' + (i++);
+              });
+              //-update variant table header count
+              var newHeader = $$('.variant.variant-title.gene-'+geneIndex)[0].firstChild.textContent.replace(/(\().+?(\))/g, "$1"+(--i)+"$2");
+              $$('.variant.variant-title.gene-'+geneIndex)[0].firstChild.textContent = newHeader;
+            }    
+          }
+        },
+        onComplete : function() {
+          deleteTrigger.disabled = false;
+        }
+      },
+      {
+         confirmationText : "$!services.localization.render('phenotips.tableMacros.rowDeleteConfirmation')"
+      });
+    },
+    ajaxAddVariantData : function (event) {
+      event.stop();
+      var addTrigger = event.element();
+      if (addTrigger.disabled) 
+         return;
+      var classname = '';
+      try {
+        var classname = /classname=([^&amp;]+)/.exec(addTrigger.href)[1];
+      } catch (err) {
+         new XWiki.widgets.Notification('Cannot add data: type not found', 'error');
```

Comment:
if this is going to be shown to the user, it should be localized. If this is an internal error, no banner should be shown to the user.